app/controllers/key_pairs_controller.rb

```diff
@@ -0,0 +1,56 @@
+class KeyPairsController < ApplicationController
+  before_action :find_compute_resource, :except => [:download]
+  before_action :find_resource, :only => [:download]
+
+  def index
+    @key_pairs = @compute_resource.get_compute_key_pairs
+  end
+
+  def download
+    @key_pair.update_attributes(audit_comment: _("%{user} Downloaded %{key} as pem file") % {user: User.current.name, key: @key_pair.name})
+    send_data @key_pair.secret, :filename => "#{@key_pair.name}.pem"
+  end
+
+  def create
+    if @compute_resource.recreate
+      process_success :success_msg => _('Successfully created'),
```

Comment:
Sometimes it is created (when no key pair), sometimes it is re-created (actually more common use)